ccl/spanconfigccl/spanconfigreconcilerccl: TestDataDriven failed

ccl/spanconfigccl/spanconfigreconcilerccl.TestDataDriven [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4463407&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4463407&tab=artifacts#/) on master @ [4a8d2f566b52fe341efe3ad60c8efe26d256a7b4](https://github.com/cockroachdb/cockroach/commits/4a8d2f566b52fe341efe3ad60c8efe26d256a7b4):


```
I220226 01:18:59.205388 24767 migration/migrationmanager/manager.go:110  [n1,intExec=set-setting,migration-mgr] 1147  no need to migrate, cluster already at newest version
I220226 01:18:59.220780 24767 util/log/event_log.go:32  [n1,intExec=set-setting] 1148 ={"Timestamp":1645838339200363822,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING version = $1","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-set-setting","PlaceholderValues":["'21.2-82'"],"SettingName":"version","Value":"21.2-82"}
I220226 01:18:59.325445 24415 3@vendor/github.com/cockroachdb/pebble/event.go:677  [n1,pebble,s1] 1149  [JOB 3] WAL created 000004
I220226 01:18:59.333820 24835 util/log/event_log.go:32  [n1,intExec=initializeClusterSecret] 1150 ={"Timestamp":1645838339284903398,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"cluster.secret\" = gen_random_uuid()::STRING","Tag":"SET CLUSTER SETTING","User":"root","ApplicationName":"$ internal-initializeClusterSecret","SettingName":"cluster.secret","Value":"d3ccb2bf-4382-46f0-8809-42c7e2560787"}
I220226 01:18:59.380971 24625 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1151 ={"Timestamp":1645838339350488208,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS defaultdb","Tag":"CREATE DATABASE","User":"root","DescriptorID":100,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"defaultdb"}
I220226 01:18:59.434511 24806 5@util/log/event_log.go:32  [n1,intExec=create-default-DB] 1152 ={"Timestamp":1645838339383581776,"EventType":"create_database","Statement":"CREATE DATABASE IF NOT EXISTS postgres","Tag":"CREATE DATABASE","User":"root","DescriptorID":102,"ApplicationName":"$ internal-create-default-DB","DatabaseName":"postgres"}
I220226 01:18:59.585472 24072 server/server_sql.go:1214  [n1] 1153  done ensuring all necessary startup migrations have run
I220226 01:18:59.587218 24948 kv/kvclient/rangefeed/rangefeedcache/watcher.go:303  [n1] 1154  tenant-settings-watcher: established range feed cache
I220226 01:18:59.591550 24865 jobs/job_scheduler.go:381  [n1] 1155  waiting 2m0s before scheduled jobs daemon start
I220226 01:18:59.595033 24947 server/auto_upgrade.go:59  [n1] 1156  no need to upgrade, cluster already at the newest version
I220226 01:18:59.764918 24388 3@vendor/github.com/cockroachdb/pebble/event.go:677  [n1,pebble,s1] 1157  [JOB 4] WAL created 000005
I220226 01:18:59.771627 24072 1@server/server_sql.go:1283  [n1] 1158  serving sql connections
I220226 01:18:59.773591 24072 testutils/testcluster/testcluster.go:1165  [-] 1159  WaitForFullReplication
I220226 01:18:59.773689 24072 testutils/testcluster/testcluster.go:1169  [-] 1160  WaitForFullReplication took: 503ns
I220226 01:18:59.773833 24072 2@rpc/context.go:1211  [n1,rnode=1,raddr=127.0.0.1:38671,class=default] 1161  dialing
I220226 01:18:59.884623 24945 util/log/event_log.go:32  [n1,client=127.0.0.1:35856,user=root] 1162 ={"Timestamp":1645838339867044197,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.rangefeed.enabled\" = true","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.rangefeed.enabled","Value":"true"}
I220226 01:18:59.900673 24945 util/log/event_log.go:32  [n1,client=127.0.0.1:35856,user=root] 1163 ={"Timestamp":1645838339891848688,"EventType":"set_cluster_setting","Statement":"SET CLUSTER SETTING \"kv.closed_timestamp.target_duration\" = '100ms'","Tag":"SET CLUSTER SETTING","User":"root","SettingName":"kv.closed_timestamp.target_duration","Value":"00:00:00.1"}
I220226 01:19:00.340661 25014 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:263  [-] 1164  established range feed over system.descriptors starting at time 1645838339.926212708,0
I220226 01:19:00.341806 25014 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:316  [-] 1165  established range feed over system.zones starting at time 1645838339.926212708,0
I220226 01:19:00.342410 25014 spanconfig/spanconfigsqlwatcher/sqlwatcher.go:411  [-] 1166  established range feed over system.protected_ts_records starting at time 1645838339.926212708,0
I220226 01:19:01.076393 24303 1@gossip/gossip.go:1510  [n1] 1167  node has connected to cluster via gossip
I220226 01:19:01.077188 24303 kv/kvserver/stores.go:269  [n1] 1168  wrote 0 node addresses to persistent storage
I220226 01:19:01.138984 24945 5@util/log/event_log.go:32  [n1,client=127.0.0.1:35856,user=root] 1169 ={"Timestamp":1645838341076465324,"EventType":"create_database","Statement":"CREATE DATABASE db","Tag":"CREATE DATABASE","User":"root","DescriptorID":104,"DatabaseName":"db"}
I220226 01:19:01.139112 24945 5@util/log/event_log.go:32  [n1,client=127.0.0.1:35856,user=root] 1170 ={"Timestamp":1645838341076465324,"EventType":"create_table","Statement":"CREATE TABLE db.public.t1 ()","Tag":"CREATE TABLE","User":"root","DescriptorID":106,"TableName":"db.public.t1"}
I220226 01:19:01.139153 24945 5@util/log/event_log.go:32  [n1,client=127.0.0.1:35856,user=root] 1171 ={"Timestamp":1645838341076465324,"EventType":"create_table","Statement":"CREATE TABLE db.public.t2 ()","Tag":"CREATE TABLE","User":"root","DescriptorID":107,"TableName":"db.public.t2"}
I220226 01:19:02.004331 25911 kv/kvserver/replica_command.go:420  [n1,split,s1,r44/1:/{Table/50-Max}] 1172  initiating a split of this range at key /Table/106 [r45] (span config)
I220226 01:19:02.014528 25924 kv/kvserver/replica_command.go:420  [n1,split,s1,r45/1:/{Table/106-Max}] 1173  initiating a split of this range at key /Table/107 [r46] (span config)
    datadriven_test.go:122: 
        /go/src/github.com/cockroachdb/cockroach/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/protectedts:58: 
        output didn't match expected:
        @@ -1,7 +1,7 @@
        -delete {entire-keyspace}
        -upsert {entire-keyspace}                   protection_policies=[{ts: 1} {ts: 5}]
         delete /Table/10{6-7}
         upsert /Table/10{6-7}                      protection_policies=[{ts: 3} {ts: 4}]
         delete /Table/10{7-8}
         upsert /Table/10{7-8}                      protection_policies=[{ts: 3} {ts: 4}]
        +delete {entire-keyspace}
        +upsert {entire-keyspace}                   protection_policies=[{ts: 1} {ts: 5}]
         
I220226 01:19:03.499859 24072 testutils/testcluster/testcluster.go:107  [-] 1174  TestCluster quiescing nodes
W220226 01:19:03.500157 24478 sql/sqlliveness/slinstance/slinstance.go:224  [n1] 1175  exiting heartbeat loop
I220226 01:19:03.500761 21979 2@rpc/context.go:817  [n1] 1176  closing [{n1: 127.0.0.1:38671 (default)} {n0: 127.0.0.1:38671 (default)}]
W220226 01:19:03.502089 26264 kv/txn.go:720  [n1,intExec=expire-sessions] 1177  failure aborting transaction: node unavailable; try another peer; abort caused by: query execution canceled
W220226 01:19:03.502380 24705 kv/txn.go:720  [n1] 1178  failure aborting transaction: node unavailable; try another peer; abort caused by: expire-sessions: context canceled
E220226 01:19:03.502450 24705 jobs/registry.go:716  [n1] 1179  error expiring job sessions: expire-sessions: context canceled
W220226 01:19:03.502508 24705 kv/txn.go:720  [n1] 1180  failure aborting transaction: node unavailable; try another peer; abort caused by: context canceled
E220226 01:19:03.502551 24705 jobs/registry.go:724  [n1] 1181  failed to serve pause and cancel requests: context canceled
W220226 01:19:03.502578 24705 jobs/registry.go:766  [n1] 1182  canceling all adopted jobs due to stopper quiescing
    --- FAIL: TestDataDriven/protectedts (5.08s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/protectedts
+++ b/pkg/ccl/spanconfigccl/spanconfigreconcilerccl/testdata/protectedts
@@ -50,6 +50,13 @@ protect record-id=4 ts=4
 descs 104
 ----
 
+mutations
+----
+delete /Table/10{6-7}
+upsert /Table/10{6-7}                      protection_policies=[{ts: 3} {ts: 4}]
+delete /Table/10{7-8}
+upsert /Table/10{7-8}                      protection_policies=[{ts: 3} {ts: 4}]
+
 # Write a protected timestamp record on the entire keyspace.
 protect record-id=5 ts=5
 cluster
@@ -59,10 +66,6 @@ mutations
 ----
 delete {entire-keyspace}
 upsert {entire-keyspace}                   protection_policies=[{ts: 1} {ts: 5}]
-delete /Table/10{6-7}
-upsert /Table/10{6-7}                      protection_policies=[{ts: 3} {ts: 4}]
-delete /Table/10{7-8}
-upsert /Table/10{7-8}                      protection_policies=[{ts: 3} {ts: 4}]
 
 state limit=3
 ----

